### DISCUSSION TITLE
SSD replacement?

### DISCUSSION THREAD

**Discussion by saifulkhaliq** [Qonfused/ASUS-ZenBook-Pro-Duo-15-UX581-Hackintosh#7] [Q&A]
Hey, I’m really happy and appreciate this project.
But there’s one thing I’m curious about: what does “requires SSD replacement” mean when the UX581GV doesn’t have an SSD slot?
Thanks in advance!

**Comment by Qonfused** [comment]
The UX581GV does include a removable M.2 SSD (just above the battery on the left side), though these models come with a Samsung PM981 drive which is [notoriously problematic for macOS](https://dortania.github.io/Anti-Hackintosh-Buyers-Guide/Storage.html).

![image](https://github.com/user-attachments/assets/e756a3ed-80db-4034-8790-f08f3d40ea07)

Unfortunately the PM981 is completely unsupported in macOS; some issues are partially fixed with NVMeFix but will still cause kernel panics that prevent booting macOS.

In my opinion, the best replacement SSDs are ones that include the Phison E16 or E18 controller (another user compiled a list of those SSDs [here](https://www.reddit.com/r/hackintosh/comments/sio8aj/comment/hvan5iy/)). There is also the WD SN750 which has an older E12 controller but is also a great budget option.
